crates/bevy_ui/src/widget/slider.rs

```diff
@@ -0,0 +1,171 @@
+            self.value = value;
+            return Ok(());
+        }
+
+        Err(SliderValueError::ValueOutOfSliderRange)
+    }
+
+    /// Retrieves the slider value
+    pub fn value(&self) -> f32 {
+        self.value
+    }
+
+    /// Retrieves the minimum slider value
+    pub fn min(&self) -> f32 {
+        self.min
+    }
+
+    /// Retrieves the maximum slider value
+    pub fn max(&self) -> f32 {
+        self.max
+    }
+
+    /// Retrieves the slider step
+    pub fn step(&self) -> f32 {
+        self.step
+    }
+}
+
+/// Error connected to setting the value of a slider
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Error)]
+pub enum SliderValueError {
+    #[error("the value given to the Slider is out of range")]
+    ValueOutOfSliderRange,
+}
+
+/// Marker struct for slider handles
+#[derive(Component, Debug, Default, Clone, Copy, Reflect)]
+#[reflect(Component, Default)]
+pub struct SliderHandle;
+
+/// Whether the slider is currently being dragged
+#[derive(Component, Debug, Default, Clone, Copy, Reflect, Deref, DerefMut)]
+#[reflect(Component, Default)]
+pub struct SliderDragged(bool);
+
+/// A label for the [`update_slider_value`] system
+#[derive(SystemLabel)]
+pub struct UpdateSliderValue;
+
+/// System for updating slider value based on the user input
+pub fn update_slider_value(
+    mouse_button_input: Res<Input<MouseButton>>,
+    touches_input: Res<Touches>,
+    mut slider_query: Query<(
+        &mut Slider,
+        &mut SliderDragged,
+        &Interaction,
+        &RelativeCursorPosition,
+    )>,
+) {
+    let mouse_released =
+        mouse_button_input.just_released(MouseButton::Left) || touches_input.any_just_released();
+
+    for (mut slider, mut slider_dragged, interaction, cursor_position) in slider_query.iter_mut() {
+        if mouse_released {
+            slider_dragged.0 = false;
+        }
+
+        if *interaction == Interaction::Clicked {
+            slider_dragged.0 = true;
+        }
+
+        if slider_dragged.0 {
+            let max = slider.max();
+            let min = slider.min();
+
+            slider
+                .set_value(cursor_position.x.clamp(0., 1.) * (max - min) + min)
+                .unwrap(); // The unwrap here is alright since the value is clamped between min and max, so it shouldn't return an error
+        }
+    }
+}
+
+/// System for updating the slider handle position based on the parent slider value
+pub fn update_slider_handle(
+    slider_query: Query<(&Slider, &Node, &Children)>,
+    mut slider_handles_query: Query<(&Node, &mut Style), With<SliderHandle>>,
+) {
+    for (slider, slider_node, slider_children) in slider_query.iter() {
+        for child in slider_children {
+            let (slider_handle_node, mut slider_handle_style) =
+                slider_handles_query.get_mut(*child).unwrap();
+
+            let slider_width = slider_node.size().x - slider_handle_node.size().x;
+
+            slider_handle_style.position.left = Val::Px(
```

Comment:
1. Using `position` instead of margin here is the correct approach. 👍 
2. I assume only works when this node's position type is set to `PositionType::Relative`?

Reply:
I tried changing it to absolute in the example and it seems to be working correctly.